channels_redis/core.py

```diff
@@ -154,12 +161,21 @@ def _setup_encryption(self, symmetric_encryption_keys):
                 # Wait on the receive buffer's contents
                 # TODO: Two coroutines rather than a poll
                 while True:
-                    if self.receive_buffer.get(channel, None):
-                        message = self.receive_buffer[channel][0]
-                        if len(self.receive_buffer[channel]) == 1:
-                            del self.receive_buffer[channel]
+
+                    messages = self.receive_buffer.get(channel, None)
```

Comment:
Why did you change this section? It looks like you're working around the single-process-channel implementation

Reply:
Yes this is part I was trying to figure out, without these changes the `test_groups_multiple_hosts*` tests fail as the receive_buffer never get populated with the message, and the `(await channel_layer.receive(channel))` times out after a group_send